src/main/java/net/dries007/tfc/objects/recipes/UnmoldRecipe.java

```diff
@@ -5,182 +5,172 @@
-                        return output;
-                    }
-                }
-            }
+            ItemStack itemstack = inv.getStackInSlot(moldSlotID);
+            IMoldHandler moldHandler = ((IMoldHandler) itemstack.getCapability(FLUID_HANDLER_CAPABILITY, null));
+            this.moldHandler = moldHandler;
+            if (moldHandler.getMetal() == null || moldHandler.getAmount() != 100 || moldHandler.isMolten())
+                return false;
         }
-        return ItemStack.EMPTY;
+
+        return canCraft;
     }
 
     @Override
-    public boolean canFit(int width, int height)
+    @Nonnull
+    public ItemStack getCraftingResult(InventoryCrafting inv)
     {
-        return true; // width * height > 1;
+        return getOutputItem(moldHandler, type);
     }
 
     @Override
-    @Nonnull
-    public ItemStack getRecipeOutput()
+    public boolean canFit(int width, int height)
     {
-        return ItemStack.EMPTY;
+        return true;
     }
 
     @Override
-    @Nonnull
-    public NonNullList<ItemStack> getRemainingItems(InventoryCrafting inv)
+    public boolean isDynamic()
+    {
+        return true;
+    }
+
+    public ItemStack getOutputItem(final IMoldHandler moldHandler, final Metal.ItemType type)
     {
-        EntityPlayer player = ForgeHooks.getCraftingPlayer();
-        if (player != null)
+        Metal m = moldHandler.getMetal();
+        if (m != null)
         {
-            if (mold.type.getMoldReturnRate() < 1 || mold.type.getMoldReturnRate() > 0)
+            ItemStack output = new ItemStack(ItemMetal.get(m, type));
+            IItemHeat heat = output.getCapability(ITEM_HEAT_CAPABILITY, null);
+            if (heat != null)
             {
-                if (Constants.RNG.nextFloat() <= mold.type.getMoldReturnRate())
-                {
-                    player.addItemStackToInventory(new ItemStack(mold));
-                }
+                heat.setTemperature(moldHandler.getTemperature());
             }
+            return output;
         }
-        return ForgeHooks.defaultRecipeGetRemainingItems(inv);
+        return ItemStack.EMPTY;
     }
 
-    @Override
-    public boolean isDynamic()
+    @SuppressWarnings("unused")
+    public static class Factory implements IRecipeFactory
     {
-        return true;
-    }
+        @Override
+        public IRecipe parse(final JsonContext context, final JsonObject json)
+        {
+            final NonNullList<Ingredient> ingredients = RecipeUtils.parseShapeless(context, json);
+            final String result = JsonUtils.getString(json, "result");
+            final Metal.ItemType type = Metal.ItemType.valueOf(result.toUpperCase());
+            final String group = JsonUtils.getString(json, "group", "");
 
-    @Override
-    @Nonnull
-    public String getGroup()
-    {
-        return MOD_ID + ":unmold_" + mold.type.name().toLowerCase();
+            //Chance of getting the mold back
+            float chance = 0;
```

Comment:
Unnecessary duplication of `Metal.ItemType.moldReturnRate`

Reply:
I was just thinking it shouldn't be hardcoded like that, but should I return it to it's original state?

Reply:
Fine with me, but one of them should be removed.

Reply:
Okay, will do!